google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java

```diff
@@ -272,9 +275,11 @@ private T performSet(
         DocumentTransform.fromFieldPathMap(documentReference, documentData);
 
     if (options.getFieldMask() != null) {
-      List<FieldPath> fieldMask = new ArrayList<>(options.getFieldMask());
-      fieldMask.removeAll(documentTransform.getFields());
-      documentMask = new FieldMask(fieldMask);
+      TreeSet<FieldPath> fieldPaths =
```

Comment:
Why is TreeSet used in this class too instead of ArrayList? I agree that the streaming API is a good improvement, but I'm not clear on the benefits of TreeSet, which could negate any performance gains.

Reply:
Nevermind. @tom-andersen pointed out to me that TreeSet was already there; this change avoids creating an intermediary collection that is (almost) immediately converted to a TreeSet then thrown away.